### PR TITLE
Fix galleries expand past the width of the screen when the links are many

### PR DESCRIPTION
**Pull request by pelumy**
## Issues covered
[#24](https://github.com/verse-pbc/issues/issues/24)

## Description
This pull request fixes an issue where galleries overflow the screen when there are many links. It updates the layout of the gallery dot indicator to ensure the dots stay within note card's bounds.

## How to test
1. Go to the discover tab and search for the following notes:
     - note1r0nwluducamvwd022t9mxq5teqnxvnwqpq6zwrudgpvg7f8f795qhall8h
     - note10n297akqfrl4kkxj282hzx2l33f3ve3446l7u7gx6pmrnsh6g3zqgec5gg
2. Please check that the gallery page indicator is not over-stretching the note card.

## Screenshots/Video

Before:

Screenshot 2024-11-15 at 5 32 33 PM

After:

https://github.com/user-attachments/assets/d25cb05f-dd2c-4b1b-ac85-716146f85fd7

### CODE REVIEW

**Review by joshuatbrown — COMMENTED** [review]
This is much better already than the overflowing galleries we have in the app today. I'm impressed -- I think this would've taken me a while to figure out. In addition to my suggestion inline, I think it would be really nice, if possible, to have the dots move and animate each time, like the iOS lock screen switcher. We can talk more about that -- it's hard to describe in a comment here.

File: Nos/Views/Components/Media/GalleryView.swift [hunk 1]
```diff
@@ -130,24 +130,59 @@ fileprivate struct GalleryIndexView: View {
     private let secondaryFill = Color.galleryIndexDotSecondary
 
     /// The scale of the circles representing tabs that aren't selected, relative to `circleSize`.
-    private let smallScale: CGFloat = 0.75
+    private let smallScale: CGFloat = 0.5
+
+    /// The maximum distance (in pages) from the selected index for visible circles.
+    /// Circles outside this range are not displayed.
+    private let maxDistance = 6
```
[the remark this answers is not in the document]
Comment (pelumy) [reply Nos/Views/Components/Media/GalleryView.swift:137]: @joshuatbrown, thank you! Is tomorrow a great time to discuss this?

[the remark this answers is not in the document]
Comment (joshuatbrown) [reply Nos/Views/Components/Media/GalleryView.swift:137]: @pelumy I'm happy to talk any time; let's discuss finding a time in Slack.

**Comment by joshuatbrown** [comment]
👀

**Review by joshuatbrown — APPROVED** [review]
This is great!

File: Nos/Views/Media/AspectRatioContainer.swift
Comment (joshuatbrown) [inline Nos/Views/Media/AspectRatioContainer.swift:1]: This and the other files below were duplicates. 🙀

**Comment by pelumy** [comment]
You did a lot of work @joshuatbrown!

**Pull request merged**